snap/info.go

```diff
@@ -224,6 +224,8 @@ type Info struct {
 	Hooks            map[string]*HookInfo
 	Plugs            map[string]*PlugInfo
 	Slots            map[string]*SlotInfo
+	ToplevelPlugs    []*PlugInfo
```

Comment:
these could be private for now right? (I tried the change locally and didn't see any issues)

Reply:
Did you run all the tests? For me making it private makes reflection fail:
```
PANIC: details_v2_test.go:186: detailsV2Suite.TestInfoFromStoreSnap

... Panic: reflect.Value.Interface: cannot return value obtained from unexported field or method (PC=0x45931A)

/usr/lib/go-1.10/src/runtime/panic.go:502
```

Reply:
ah, no, but that should be fixable let me try

Reply:
something like this:

```
diff --git a/store/details_v2_test.go b/store/details_v2_test.go
index 791cbd223..085ad48b7 100644
--- a/store/details_v2_test.go
+++ b/store/details_v2_test.go
@@ -290,8 +290,6 @@ func (s *detailsV2Suite) TestInfoFromStoreSnap(c *C) {
                "Apps",
                "LegacyAliases",
                "Hooks",
-               "ToplevelPlugs",
-               "ToplevelSlots",
                "BadInterfaces",
                "Broken",
                "MustBuy",
@@ -306,6 +304,10 @@ func (s *detailsV2Suite) TestInfoFromStoreSnap(c *C) {
                t := x.Type()
                for i := 0; i < x.NumField(); i++ {
                        f := t.Field(i)
+                       if f.PkgPath != "" {
+                               // not exported, ignore
+                               continue
+                       }
                        v := x.Field(i)
                        if f.Anonymous {
                                checker(pfx+f.Name+".", v)
```